src/shared/api/services/fetchData.ts

```diff
@@ -0,0 +1,7 @@
+import { axiosInstance } from './axiosInstance';
+import type { AxiosResponse } from 'axios';
+
+export const fetchData = async <T>(url: string): Promise<T> => {
```

Comment:
generic added for my teammates , after invoke this function we can use generic to type result data that we will get and use